torch/lib/TH/THTensorDimApply.h

```diff
@@ -1,7 +1,32 @@
 #ifndef TH_TENSOR_DIM_APPLY_INC
 #define TH_TENSOR_DIM_APPLY_INC
 
-#define TH_TENSOR_DIM_APPLY3(TYPE1, TENSOR1, TYPE2, TENSOR2, TYPE3, TENSOR3, DIMENSION, CODE) \
+#define TH_TENSOR_DIM_APPLY3_SIZE_EQ_EXCEPT_DIM(TENSOR1, TENSOR2, TENSOR3, DIMENSION) \
```

Comment:
Can you add a comment saying how this macro differs from all other applies? We have so many variants that I don't even know what they're for at this point.

Reply:
Sure, I will add one soon. 

As a quick explanation, this is a part of original `TH_TENSOR_DIM_APPLY3`, extracted out meant to refactor the size check so that scatter can apply its own check. The `TH_TENSOR_DIM_APPLY3` is still there below.